src/manifest.c

```diff
@@ -934,7 +934,7 @@ void link_submanifests(struct manifest *m1, struct manifest *m2, struct list *su
 
 /* if component is specified explicitly, pull in submanifest only for that
  * if component is not specified, pull in any tracked component submanifest */
-struct list *recurse_manifest(struct manifest *manifest, struct list *subs, const char *component)
+struct list *recurse_manifest(struct manifest *manifest, struct list *subs, const char *component, bool server)
```

Comment:
I'm not very happy with the current state of this function. For example, if `component` is non-NULL, it will call `load_manifest` once, and that's it. Simpler would be to wrap `load_manifest` differently so that the API is less confusing. At which point, we could remove the `component` parameter entirely, and the code would become much more clear (IMO).

That said, leave it as-is for now, but I would strongly consider refactoring this function (and possibly renaming it) for conciseness and clarity.

Reply:
I agree, this should be fixed up in the future.

Reply:
Okay, sounds good.